server/cherry_pick.go

```diff
@@ -56,14 +56,14 @@ func (s *Server) checkIfNeedCherryPick(pr *model.PullRequest) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
 	defer cancel()
 
-	prCherryCandidate, _, err := s.GithubClient.PullRequests.Get(ctx, pr.RepoOwner, pr.RepoName, pr.Number)
-	if err != nil {
-		mlog.Error("Error getting the PR info", mlog.Err(err))
+	if !pr.Merged.Valid || !pr.Merged.Bool {
+		mlog.Info("PR not merged, not cherry picking", mlog.Int("PR Number", pr.Number), mlog.String("Repo", pr.RepoName))
 		return
 	}
 
-	if !prCherryCandidate.GetMerged() {
-		mlog.Info("PR not merged, not cherry picking", mlog.Int("PR Number", prCherryCandidate.GetNumber()), mlog.String("Repo", pr.RepoName))
+	prCherryCandidate, _, err := s.GithubClient.PullRequests.Get(ctx, pr.RepoOwner, pr.RepoName, pr.Number)
```

Comment:
We need to think about the disparity between model.PullRequest and github.PullRequest a bit more. Just because that field is not there, we have to query GH again and waste those tokens.

Reply:
+1

Reply:
I opened this PR to fix the immediate issue we were seeing in the logs.
 
The follow-up ticket (https://mattermost.atlassian.net/browse/MM-27487) tracks the effort of updating all the models, and bringing this change to the other objects.

Reply:
Oh it was a general observation, and not meant to be done in this ticket. Sorry if it implied otherwise.

Reply:
>Oh it was a general observation, and not meant to be done in this ticket. Sorry if it implied otherwise.

No worries. I didn't infer it that way :) 

I've linked this PR in the Jira ticket, so the lucky dev fixing that issue has context.